app/services/healthcheck/pvb_api_check.rb

```diff
@@ -4,8 +4,18 @@ class PvbApiCheck
 
     def initialize(description)
       build_report(description) do
-        { ok: PrisonVisits::Api.instance.healthy? }
+        { ok: healthy_pvb_connection }
       end
     end
+
+  private
+
+    def healthy_pvb_connection
+      client.healthcheck.status == 200
+    end
+
+    def client
+      PrisonVisits::Client.new(Rails.configuration.api_host, persistent: false)
```

Comment:
This might make the 502s problem worth, non persistent connection will make each request longuer, bare in made that we already receive 3 simultaneous health check request per instance.